compiler/rustc_passes/src/check_attr.rs

```diff
@@ -989,16 +989,36 @@ impl CheckAttrVisitor<'tcx> {
 
         for hint in &hints {
             let (article, allowed_targets) = match hint.name_or_empty() {
-                _ if !matches!(target, Target::Struct | Target::Enum | Target::Union) => {
-                    ("a", "struct, enum, or union")
+                _ if !matches!(
+                    target,
+                    Target::Struct | Target::Enum | Target::Union | Target::Fn
+                ) =>
+                {
+                    ("a", "struct, enum, function, or union")
                 }
-                name @ sym::C | name @ sym::align => {
-                    is_c |= name == sym::C;
+                sym::C => {
+                    is_c = true;
                     match target {
                         Target::Struct | Target::Union | Target::Enum => continue,
                         _ => ("a", "struct, enum, or union"),
                     }
                 }
+                sym::align => {
+                    if let (Target::Fn, true) = (target, !self.tcx.features().fn_align) {
+                        feature_err(
+                            &self.tcx.sess.parse_sess,
+                            sym::fn_align,
+                            hint.span(),
+                            "the attribute `repr(align(...))` on functions is unstable",
```

Comment:
```suggestion
                            "`repr(align)` attributes on functions are unstable",
```
nit: copying the way we write this in https://github.com/rust-lang/rust/blob/6523b721c3d8a92cf06aab04dd15ebe7fd5a9fb7/compiler/rustc_attr/src/builtin.rs#L924